cli/cmd/createCluster.go

```diff
@@ -21,6 +27,219 @@ ksctl create-cluster ["azure", "gcp", "aws", "local"]
+		if !isSet {
+			logger.Verbose = false
+		}
+
+		payload := civo.CivoProvider{
+			ClusterName: clusterName,
+			Region:      region,
+			Application: apps,
+			CNIPlugin:   cni,
+			HACluster:   false,
+			Spec: utils.Machine{
+				Disk:         nodeSize,
+				ManagedNodes: noManagedNodes,
+			},
+		}
+		err := payload.CreateCluster(logger)
+		if err != nil {
+			logger.Err(err.Error())
+			return
+		}
+		logger.Info("CREATED CLUSTER")
+	},
+}
+
+var createClusterLocal = &cobra.Command{
+	Use:   "local",
+	Short: "Use to create a LOCAL cluster in Docker",
+	Long: `It is used to create cluster with the given name from user. For example:
+
+ksctl create-cluster local <arguments to civo cloud provider>
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		isSet := cmd.Flags().Lookup("verbose").Changed
+		logger := log.Logger{Verbose: true}
+		if !isSet {
+			logger.Verbose = false
+		}
+
+		cargo := local.ClusterInfoInjecter(clusterName, noManagedNodes)
+		logger.Info("Building cluster", "")
+		if err := local.CreateCluster(logger, cargo); err != nil {
+			logger.Err(err.Error())
+			return
+		}
+		logger.Info("CREATED CLUSTER")
+	},
+}
+
+var createClusterHACivo = &cobra.Command{
+	Use:   "ha-civo",
+	Short: "Use to create a HA CIVO k3s cluster",
+	Long: `It is used to create cluster with the given name from user. For example:
+
+ksctl create-cluster ha-civo <arguments to civo cloud provider>
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		isSet := cmd.Flags().Lookup("verbose").Changed
+		logger := log.Logger{Verbose: true}
+		if !isSet {
+			logger.Verbose = false
+		}
+		payload := civo.CivoProvider{
+			ClusterName: clusterName,
+			Region:      region,
+			HACluster:   true,
+			Spec: utils.Machine{
+				Disk:                nodeSize,
+				HAControlPlaneNodes: noControlPlaneNodes,
+				HAWorkerNodes:       noWorkerNodes,
+			},
+		}
+		err := payload.CreateCluster(logger)
+		if err != nil {
+			logger.Err(err.Error())
+			return
+		}
+		logger.Info("CREATED CLUSTER")
+	},
+}
+
+var createClusterHAAzure = &cobra.Command{
+	Use:   "ha-azure",
+	Short: "Use to create a HA k3s cluster in Azure",
+	Long: `It is used to create cluster with the given name from user. For example:
+
+	ksctl create-cluster ha-azure <arguments to civo cloud provider>
+	`,
+	Run: func(cmd *cobra.Command, args []string) {
+		isSet := cmd.Flags().Lookup("verbose").Changed
+		logger := log.Logger{Verbose: true}
+		if !isSet {
+			logger.Verbose = false
+		}
+
+		payload := &azure.AzureProvider{
+			ClusterName: clusterName,
+			HACluster:   true,
+			Region:      region,
+			Spec: util.Machine{
+				Disk:                nodeSize,
+				HAControlPlaneNodes: noControlPlaneNodes,
+				HAWorkerNodes:       noWorkerNodes,
+			},
+		}
+		err := payload.CreateCluster(logger)
+		if err != nil {
+			logger.Err(err.Error())
+			return
+		}
+		logger.Info("CREATED CLUSTER")
+	},
+}
+
 func init() {
 	rootCmd.AddCommand(createClusterCmd)
```

Comment:
have you given a thought of shifting these command var inside a func for better control over the command and with that we could get reusablity of that command, we can test it etc?

example: 
```
rootCmd.AddCommand(CreateClusterCmd())

func CreateClusterCmd() *cobra.Command {}
```

Reply:
not given a thought but thought it will be good if we can merge them into single file so that once if we are going to control like you mentioned above we can do it much easier

Reply:
location doesnot matter for me I meant should we make commands come via func and not var

Reply:
yes that would be great